diff --git a/includes/media.filter.inc b/includes/media.filter.inc
index 32424df..3f06bf2 100644
--- a/includes/media.filter.inc
+++ b/includes/media.filter.inc
@@ -410,6 +410,14 @@ function media_get_file_without_label($file, $view_mode, $settings = array()) {
     }
   }
 
+  // fix for issue #1411340 "Resizing images in WYSIWYG"
+  if (isset($element['#attributes']) && isset($element['#attributes']['width'])) {
+    $element['#width'] = $element['#attributes']['width'];
+  }
+  if (isset($element['#attributes']) && isset($element['#attributes']['height'])) {
+    $element['#height'] = $element['#attributes']['height'];
+  }
+
   return $element;
 }
 
